packages/react/src/css-map/pseudos.ts

```diff
@@ -0,0 +1,143 @@
+// List of pseudo-classes and pseudo-elements are from csstype
+// but with & added in the front, so that we target the current element
+// (instead of a child element)
+
+type AdvancedPseudos =
+  | '&:-moz-any()'
+  | '&:-moz-dir'
+  | '&:-webkit-any()'
+  | '&::cue'
+  | '&::cue-region'
+  | '&::part'
+  | '&::slotted'
+  | '&::view-transition-group'
+  | '&::view-transition-image-pair'
+  | '&::view-transition-new'
+  | '&::view-transition-old'
+  | '&:dir'
+  | '&:has'
+  | '&:host'
+  | '&:host-context'
+  | '&:is'
+  | '&:lang'
+  | '&:matches()'
+  | '&:not'
+  | '&:nth-child'
+  | '&:nth-last-child'
+  | '&:nth-last-of-type'
+  | '&:nth-of-type'
+  | '&:where';
+
+type SimplePseudos =
+  | '&:-khtml-any-link'
+  | '&:-moz-any-link'
+  | '&:-moz-focusring'
+  | '&:-moz-full-screen'
+  | '&:-moz-placeholder'
+  | '&:-moz-read-only'
+  | '&:-moz-read-write'
+  | '&:-moz-ui-invalid'
+  | '&:-moz-ui-valid'
+  | '&:-ms-fullscreen'
+  | '&:-ms-input-placeholder'
+  | '&:-webkit-any-link'
+  | '&:-webkit-full-screen'
+  | '&::-moz-placeholder'
+  | '&::-moz-progress-bar'
+  | '&::-moz-range-progress'
+  | '&::-moz-range-thumb'
+  | '&::-moz-range-track'
+  | '&::-moz-selection'
+  | '&::-ms-backdrop'
+  | '&::-ms-browse'
+  | '&::-ms-check'
+  | '&::-ms-clear'
+  | '&::-ms-expand'
+  | '&::-ms-fill'
+  | '&::-ms-fill-lower'
+  | '&::-ms-fill-upper'
+  | '&::-ms-input-placeholder'
+  | '&::-ms-reveal'
+  | '&::-ms-thumb'
+  | '&::-ms-ticks-after'
+  | '&::-ms-ticks-before'
+  | '&::-ms-tooltip'
+  | '&::-ms-track'
+  | '&::-ms-value'
+  | '&::-webkit-backdrop'
+  | '&::-webkit-input-placeholder'
+  | '&::-webkit-progress-bar'
+  | '&::-webkit-progress-inner-value'
+  | '&::-webkit-progress-value'
+  | '&::-webkit-slider-runnable-track'
+  | '&::-webkit-slider-thumb'
+  | '&::after'
+  | '&::backdrop'
+  | '&::before'
+  | '&::cue'
+  | '&::cue-region'
+  | '&::first-letter'
+  | '&::first-line'
+  | '&::grammar-error'
+  | '&::marker'
+  | '&::placeholder'
+  | '&::selection'
+  | '&::spelling-error'
+  | '&::target-text'
+  | '&::view-transition'
+  | '&:active'
```

Comment:
Would be great to reduce the total amount of declarations. So instead of allowing both ::active and :active we settle on one.

Reply:
Ah good catch, didn't notice the duplicated pseudo-elements (:before, :after, :first-letter, :first-line)